components/collator/tests/tests.rs

```diff
@@ -1007,16 +1009,162 @@ fn test_lt_primary() {
         check_expectations(&collator, &left, &right, &expectations);
     }
 }
+#[test]
+fn test_fi() {
+    // Adapted from ficoll.cpp in ICU4C
+    // Testing that w and v behave as in the root collation is for checking
+    // that the sorting collation doesn't exhibit the behavior of the search
+    // collation, which (somewhat questionably) treats w and v as primary-equal.
+    let left = [
+        "wat",
+        "vat",
+        "aübeck",
+        "Låvi",
+        // ICU4C has a duplicate of the case below.
+        // The duplicate is omitted here.
+        // Instead, the subsequent tests are added for ICU4X.
+        "ä",
+        "a\u{0308}",
+    ];
+    let right = ["vat", "way", "axbeck", "Läwe", "o", "ä"];
+    let expectation = [
+        Ordering::Greater,
+        Ordering::Less,
+        Ordering::Greater,
+        Ordering::Less,
+        Ordering::Greater,
+        Ordering::Equal,
+    ];
+    let locale: DataLocale = langid!("fi").into();
+    let mut options = CollatorOptions::new();
+
+    options.strength = Some(Strength::Tertiary);
+    let collator = Collator::try_new(&locale, options).unwrap();
+    check_expectations(&collator, &left, &right, &expectation);
+
+    options.strength = Some(Strength::Primary);
+    let collator = Collator::try_new(&locale, options).unwrap();
+    check_expectations(&collator, &left, &right, &expectation);
+}
+
+#[test]
+fn test_sv() {
+    // This is the same as test_fi. The purpose of this copy is to test that
+    // Swedish defaults to "reformed", which behaves like Finnish "standard",
+    // and not to "standard", which behaves like Finnish "traditional".
+
+    // Adapted from ficoll.cpp in ICU4C
+    // Testing that w and v behave as in the root collation is for checking
+    // that the sorting collation doesn't exhibit the behavior of the search
+    // collation, which (somewhat questionably) treats w and v as primary-equal.
+
+    let left = [
+        "wat",
+        "vat",
+        "aübeck",
+        "Låvi",
+        // ICU4C has a duplicate of the case below.
+        // The duplicate is omitted here.
+        // Instead, the subsequent tests are added for ICU4X.
+        "ä",
+        "a\u{0308}",
+    ];
+    let right = ["vat", "way", "axbeck", "Läwe", "o", "ä"];
+    let expectations = [
+        Ordering::Greater,
+        Ordering::Less,
+        Ordering::Greater,
+        Ordering::Less,
+        Ordering::Greater,
+        Ordering::Equal,
+    ];
+    let locale: DataLocale = langid!("sv").into();
+    let mut options = CollatorOptions::new();
+
+    options.strength = Some(Strength::Tertiary);
+    let collator = Collator::try_new(&locale, options).unwrap();
+    check_expectations(&collator, &left, &right, &expectations);
+
+    options.strength = Some(Strength::Primary);
+    let collator = Collator::try_new(&locale, options).unwrap();
+    check_expectations(&collator, &left, &right, &expectations);
+}
+
+#[test]
+fn test_nb_nn_no() {
```

Comment:
Issue: Your test coverage is not as high as it was before; you deleted these tests:

```rust
    // Test "no" macro language without fallback (should equal expected)
    let collator =
        Collator::try_new_unstable(&provider, &locale!("no").into(), CollatorOptions::new())
            .unwrap();
    let mut strs = input.clone();
    strs.sort_by(|a, b| collator.compare(a, b));
    assert_eq!(strs, expected);

    // Test "und" without fallback (should NOT equal expected)
    let collator =
        Collator::try_new_unstable(&provider, &locale!("und").into(), CollatorOptions::new())
            .unwrap();
    let mut strs = input.clone();
    strs.sort_by(|a, b| collator.compare(a, b));
    assert_ne!(strs, expected);

    // Test "nb" without fallback (should fail to load)
    if Collator::try_new_unstable(&provider, &locale!("nb").into(), CollatorOptions::new()).is_ok()
    {
        panic!("Should fail to create 'nb' without fallback enabled")
    }
```

Reply:
* We don't have a provider without fallback
* This tests fallback, not collator or datagen logic. Fallback is tested somewhere else

Reply:
ok.